src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -560,46 +319,110 @@ CryptoNative_SslCtxSetCertVerifyCallback(SSL_CTX* ctx, SslCtxSetCertVerifyCallba
     SSL_CTX_set_cert_verify_callback(ctx, callback, arg);
 }
 
-// delimiter ":" is used to allow more than one strings
-// below string is corresponding to "AllowNoEncryption"
-#define SSL_TXT_Separator ":"
-#define SSL_TXT_Exclusion "!"
-#define SSL_TXT_AllIncludingNull SSL_TXT_ALL SSL_TXT_Separator SSL_TXT_eNULL
-#define SSL_TXT_NotAnon SSL_TXT_Separator SSL_TXT_Exclusion SSL_TXT_aNULL
-
 int32_t CryptoNative_SetEncryptionPolicy(SSL_CTX* ctx, EncryptionPolicy policy)
 {
-    const char* cipherString = NULL;
-    bool clearSecLevel = false;
-
     switch (policy)
     {
+        case AllowNoEncryption:
+        case NoEncryption:
+            // No minimum security policy, same as OpenSSL 1.0
+            SSL_CTX_set_security_level(ctx, 0);
+            return true;
         case RequireEncryption:
-            cipherString = SSL_TXT_ALL SSL_TXT_NotAnon;
-            break;
+            return true;
+    }
 
-        case AllowNoEncryption:
-            cipherString = SSL_TXT_AllIncludingNull;
-            clearSecLevel = true;
-            break;
+    return false;
+}
 
-        case NoEncryption:
-            cipherString = SSL_TXT_eNULL;
-            clearSecLevel = true;
-            break;
+int32_t CryptoNative_SetCiphers(SSL_CTX* ctx, const char* cipherList, const char* cipherSuites)
+{
+    int32_t ret = true;
+
+    // for < TLS 1.3
+    if (cipherList != NULL)
+    {
+        ret &= SSL_CTX_set_cipher_list(ctx, cipherList);
+        if (!ret)
+        {
+            return ret;
+        }
+    }
+
+    // for TLS 1.3
+#if HAVE_OPENSSL_SET_CIPHERSUITES
+    if (CryptoNative_Tls13Supported() && cipherSuites != NULL)
+    {
+        ret &= SSL_CTX_set_ciphersuites(ctx, cipherSuites);
     }
+#else
+    (void)cipherSuites;
```

Comment:
Should this be `ret &= (cipherSuites == NULL);`?  (aka fail if cipherSuites made it this far?)

Reply:
the code was initially written with working on lower than 1.1.1 in mind (the only issue with that is we would have to have IANA -> OpenSSL string mapping somewhere since SSL_CIPHER_find is not available), assuming we got here and TLS 1.3 is not supported then setting 1.3 cipher suites will have no effect anyway since we already would have failed when setting < 1.3 cipher suites.